src/config.rs

```diff
@@ -228,24 +228,47 @@ pub struct PutOptions {
     /// database entry, the TTL for the most recent entry will be canonical.
     ///
     /// Default: the TTL configured in DbOptions when opening a SlateDB session
-    pub ttl: Option<Duration>,
+    pub ttl: Ttl,
 }
 
 impl PutOptions {
     const fn default() -> Self {
-        Self { ttl: None }
+        Self { ttl: Ttl::Default }
     }
 
-    pub(crate) fn expire_ts_from(&self, now: i64) -> Option<i64> {
-        self.ttl.map(|ttl| {
-            now + i64::try_from(ttl.as_millis()).expect(
-                "Duration could not be converted into an i64 timestamp. \
-             Perhaps the duration in millis exceeds the maximum i64?",
-            )
-        })
+    pub(crate) fn expire_ts_from(&self, default: Option<u64>, now: i64) -> Option<i64> {
+        match self.ttl {
+            Ttl::Default => match default {
+                None => None,
+                Some(default_ttl) => Self::checked_expire_ts(now, default_ttl),
+            },
+            Ttl::NoExpiry => None,
+            Ttl::ExpireAfter(ttl) => Self::checked_expire_ts(now, ttl),
+        }
+    }
+
+    fn checked_expire_ts(now: i64, ttl: u64) -> Option<i64> {
+        // for overflow, we will just assume no TTL
+        if ttl > i64::MAX as u64 {
+            return None;
+        };
+        let expire_ts = now + (ttl as i64);
+        if expire_ts < now {
+            return None;
+        };
+
+        Some(expire_ts)
     }
 }
 
+#[derive(Clone, Default)]
+pub enum Ttl {
+    #[default]
+    Default,
+    NoExpiry,
+    ExpireAfter(u64),
+}
```

Comment:
@criccomini I changed the API to better account for the distinction between "default" "no expiry" and "expire after". Cassandra, for example, uses a TTL of `0` to indicate "no expiry" but I think that's pretty confusing... LMKWYT

Reply:
I like this. I also agree 0 shouldn't be "no expiry". My intuition with "0" is that it should expire the next time compaction sees it. I like having that ability by setting it to ExpireAfter(0). Cassandra, you'd have to emulate that with a "1", which is kinda lame.